### PR TITLE
[auto_trainer] Update documentation and improve the current kwargs mechanism

### PR DESCRIPTION
**Pull request by yonishelach**
# Fixed issues
- Changed `model_class` parameter to mandatory.
- The `label_columns` param is None by default, it is not required when passing the dataset as a FeatureVector. An error message is added in the other cases. mandatory when dataset is not a FeatureVector.
- added `kwargs` instead of using the `context.parameters`, and parsed into the functions.

# Docstring improvements 
- Added explanation about the relation  between `train_test_split_size` and `test_set`.
- Added explanation about the`random_state` parameter.
- Added explanation on how to use the new `kwargs`, with the prefixes.
- Also added a use example to the docstring.

[Jira ticket](https://jira.iguazeng.com/browse/ML-5231)

### CODE REVIEW

**Comment by aviaIguazio** [comment]
@yonishelach this is why the test is stuck

[pipenv.exceptions.InstallError]:   note: This error originates from a subprocess, and is likely not a problem with pip.
[pipenv.exceptions.InstallError]:   ERROR: Failed building wheel for lightgbm
[pipenv.exceptions.InstallError]: ERROR: Could not build wheels for lightgbm, which is required to install pyproject.toml-based projects
ERROR: Couldn't install package: {}
 Package installation failed...

File: auto_trainer/function.yaml [hunk 1]
```diff
@@ -46,37 +59,35 @@ spec:
...
+          Notice that here we only pass integer values.'
         default: null
       - name: labels
         type: dict
         doc: Labels to log with the model
         default: null
       outputs:
       - default: ''
-      lineno: 110
+      lineno: 116
```
Comment (theSaarco) [inline auto_trainer/function.yaml:90]: How come we don't have `has_kwargs` for the function, now that we added `**kwargs` to the signature?
Same is true for the rest of the entrypoints in the file.

**Review by theSaarco — CHANGES_REQUESTED** [review]
I'm missing `has_kwargs` for the entrypoint.

File: auto_trainer/function.yaml (hunk 1 above)
Comment (yonishelach) [reply auto_trainer/function.yaml]: It was not available in a stable version, was resolved with one of 1.6.0 rc's

**Review by theSaarco — APPROVED** [review]
👍 sababa

**Pull request merged**